src/properties/degrees.c

```diff
@@ -517,3 +517,138 @@ igraph_error_t igraph_sort_vertex_ids_by_degree(const igraph_t *graph,
     }
     return IGRAPH_SUCCESS;
 }
+
+/**
+ * \function igraph_construct_jdm
+ * \brief Constructs a joint degree matrix.
+ *
+ * The joint degree matrix of a graph contains the number of edges, or the sum of the weights, between vertices of
+ * degree i and degree j for every (i, j). The function populates a joint degree matrix and works on directed
+ * and undirected graphs. Self loops are only counted once. If either \p dout or \p din are <= 0, the JDM
+ * will be resized automatically in the dimensions of the max degree of the vertices in the graph.
+ *
+ * \param graph A pointer to an initialized graph object.
+ * \param jdm A pointer to an initialized integer matrix that will be resized. The values will be written here.
+ * \param dout An integer for the dimension of the matrix. If \p dout and \p din are greater than 0, the matrix will be
+ *        resized do dimensions (\p dout, \p din).
+ * \param din An integer for the dimension of the matrix. If \p dout and \p din are greater than 0, the matrix will be
+ *        resized do dimensions (\p dout, \p din).
+ * \param weights An initialized integer vector containing the weights of the edges. NULL if the JDM should sum
+ *        the number of edges rather than sum up the weights.
+ * \return Error code.
+ *
+ * Time complexity: O(E),
+ * where E is the number of edges in input graph.
+ */
+
+igraph_error_t igraph_construct_jdm(const igraph_t* graph,
+                                    igraph_matrix_int_t* jdm,
+                                    igraph_integer_t dout,
+                                    igraph_integer_t din,
+                                    const igraph_vector_int_t* weights) {
+    igraph_eit_t eit;
+    igraph_es_t es;
+    igraph_integer_t no_of_edges = igraph_ecount(graph);
```

Comment:
```suggestion
    igraph_integer_t no_of_nodes = igraph_vcount(graph);
    igraph_integer_t no_of_edges = igraph_ecount(graph);
```

Let's add this now, as it will be useful for later comments I'll make.